services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -227,23 +230,33 @@ def location_from_id(cls, location_id : str):
                                     raw_file_path=d.raw_file_path,
                                     display_file_path=d.display_file_path)
                             await conn.execute(query)
+                            clean_data.append(d)
+
+                data = clean_data
+                for d in data:
+                    logger.info(d)
 
+                # same as above, make sure file is physically present on s3
+                clean_data = []
                 # MaG: This is inefficient: Do this automatically when file is modified
                 _loop = asyncio.get_event_loop()
                 session = aiobotocore.get_session(loop=_loop)
                 async with session.create_client('s3', endpoint_url="http://"+self.s3_client.endpoint, aws_access_key_id=self.s3_client.access_key,
                      aws_secret_access_key=self.s3_client.secret_key) as client:
                     responses = await asyncio.gather(*[client.list_objects_v2(Bucket=d.bucket_name, Prefix=_d) for _d in [__d.object_name for __d in data]])
                     for d, resp in zip(data, responses):
-                        d.file_size = resp['Contents'][0]['Size']
-                        d.last_modified = str(resp['Contents'][0]['LastModified'])
-                        async with self.engine.acquire() as conn:
-                            query = file_meta_data.update().\
-                            where(and_(file_meta_data.c.node_id==d.node_id,
-                                    file_meta_data.c.user_id==d.user_id)).\
-                            values(file_size=d.file_size,
-                                    last_modified=d.last_modified)
-                            await conn.execute(query)
+                        if 'Contents' in resp:
+                            clean_data.append(d)
+                            d.file_size = resp['Contents'][0]['Size']
+                            d.last_modified = str(resp['Contents'][0]['LastModified'])
+                            async with self.engine.acquire() as conn:
+                                query = file_meta_data.update().\
+                                where(and_(file_meta_data.c.node_id==d.node_id,
+                                        file_meta_data.c.user_id==d.user_id)).\
+                                values(file_size=d.file_size,
+                                        last_modified=d.last_modified)
+                                await conn.execute(query)
+                data = clean_data
```

Comment:
do you really need this last assignment?

Reply:
yes, because there is a filter belwo